src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexDeltaMassesGenerator.cpp

```diff
@@ -224,86 +249,112 @@ namespace OpenMS
+        mass_pattern_list_.push_back(doublet1);
+
+        MultiplexDeltaMasses doublet2;
+        doublet2.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[1]);
+        doublet2.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[2]);
+        mass_pattern_list_.push_back(doublet2);
+
+        MultiplexDeltaMasses doublet3;
+        doublet3.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[0]);
+        doublet3.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[2]);
+        mass_pattern_list_.push_back(doublet3);
       }
       
       // add singlets
-      std::vector<double> singlet(1, 0);
-      mass_pattern_list_.push_back(MultiplexDeltaMasses(singlet));
+      MultiplexDeltaMasses dm;
+      dm.getDeltaMasses().push_back(MultiplexDeltaMasses::DeltaMass(0,"any_label_set"));    // There are three singlets with different label sets. But only a single singlet with "any_label_set" is added.
+      mass_pattern_list_.push_back(dm);
     }
     else if (n == 4)
     {
       for (unsigned i = 0; i < m; ++i)
       {
         // add triplets
-        std::vector<double> triplet1(1, 0);
-        triplet1.push_back(mass_pattern_list_[i].getMassShiftAt(2) - mass_pattern_list_[i].getMassShiftAt(1));
-        triplet1.push_back(mass_pattern_list_[i].getMassShiftAt(3) - mass_pattern_list_[i].getMassShiftAt(1));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(triplet1));
-
-        std::vector<double> triplet2(1, 0);
-        triplet2.push_back(mass_pattern_list_[i].getMassShiftAt(2) - mass_pattern_list_[i].getMassShiftAt(0));
-        triplet2.push_back(mass_pattern_list_[i].getMassShiftAt(3) - mass_pattern_list_[i].getMassShiftAt(0));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(triplet2));
-
-        std::vector<double> triplet3(1, 0);
-        triplet3.push_back(mass_pattern_list_[i].getMassShiftAt(1) - mass_pattern_list_[i].getMassShiftAt(0));
-        triplet3.push_back(mass_pattern_list_[i].getMassShiftAt(2) - mass_pattern_list_[i].getMassShiftAt(0));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(triplet3));
+        MultiplexDeltaMasses triplet1;
+        triplet1.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[1]);
+        triplet1.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[2]);
+        triplet1.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[3]);
+        mass_pattern_list_.push_back(triplet1);
+        
+        MultiplexDeltaMasses triplet2;
+        triplet2.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[0]);
+        triplet2.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[2]);
+        triplet2.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[3]);
+        mass_pattern_list_.push_back(triplet2);
+
+        /*MultiplexDeltaMasses triplet3;
```

Comment:
what about this commented code? remove? if not please add comment what this is about